pywbem_mock/_dmtf_cim_schema.py

```diff
@@ -0,0 +1,460 @@
+unzips the MOF files into a subdirectory and makes this subdirectory
+available as a property of the :class:`~pywbem_mock.DMTFCIMSchema` object.
+
+The implementation of this class also depends on all of the qualiifiers and
+classes in the CIM schema having unique class names; this is always true for
+DMTF CIM schema releases to assure that they can be installed in the same
+namespace in a WBEM server repository.
+
+Once a DMTF CIM schema is extracted into the individual MOF files it consumes
+considerable space since it expands to almost 3000 files.  Therefore it is
+logical to remove all of the individual MOF files when not being used and also
+if the schema information is to be committed to be used in testing in an
+environment like github. Therefore, a method
+:meth:`~pywbem_mock.DMTFCIMSchema.clean` removes all of the MOF files from
+local storage.  They are restored the next time the
+:class:`~pywbem_mock.DMTFCIMSchema` object for the same version is constructed.
+
+The DMTF maintains two versions of each released schema:
+
+* Final - This schema does not contain any of the schema classes that are
+  marked experimentel.  It is considered the stable release.
+
+* Experimental - Contains the classes in the final schema plus other
+  classes that are considered experimental. It is considered less stable and
+  the experimental classes with the Experimental qualifier are subject to
+  incompatible changes .
+
+:class:`~pywbem_mock.DMTFCIMSchema` will install either the final or
+experimental schema depending on the value of the `use-experimental`
+parameter.
+
+Because it is usually necessary to compile only a subset of the DMTF CIM
+classes into a mock repository for any test, the
+:class:`~pywbem_mock.DMTFCIMSchema` class provides a
+:meth:`~pywbem_mock.build_schema_mof` method to create a MOF include file that
+includes only a subset of the classes in the downloaded DMTF CIM Schema. The
+task of listing classes to be compiled is futher simplified because the pywbem
+MOF compiler searches the DMTF schema for prerequisite classes (superclasses,
+reference classes, and EmbeddedInstance classes) so that generally only the
+leaf CIM classes required for the repository need to be in the class list. This
+speeds up the process of compiling DMTF CIM classes for any test significantly.
+"""
+import os
+from zipfile import ZipFile
+import shutil
+import six
+
+if six.PY2:
+    # pylint: disable=wrong-import-order
+    from urllib2 import urlopen
+else:
+    # pylint: disable=wrong-import-order
+    from urllib.request import urlopen
+
+
+__all__ = ['DMTFCIMSchema']
+
+
+class DMTFCIMSchema(object):
+    """
+    :class:`DMTFCIMSchema` represents a DMTF CIM Schema downloaded from the
+    DMTF web site.
+
+    This class manages the download of the DMTF schema zip file and extraction
+    of MOF files of DMTF CIM schema releases into a directory that can be used
+    by :class:`pywbem_mock.FakedWBEMConnection` to compile a mock repository
```

Comment:
Add tilde to link target to suppress the namespace.

Reply:
DONE